src/Managed.hs

```diff
@@ -0,0 +1,35 @@
+module Managed where
+
+import Lookup
+import Obj
+import Types
+
+-- | Find out if a type is "external", meaning it is not defined by the user
+--   in this program but instead imported from another C library or similar.
+-- NOTE: Quite possibly this function should be removed and we should rely on 'isManaged' instead?
+isExternalType :: TypeEnv -> Ty -> Bool
+isExternalType typeEnv (PointerTy p) =
+  isExternalType typeEnv p
+isExternalType typeEnv (StructTy (ConcreteNameTy name) _) =
+  case lookupBinder (SymPath [] name) (getTypeEnv typeEnv) of
+    Just (Binder _ (XObj (Lst (XObj (ExternalType _) _ _ : _)) _ _)) -> True
+    Just _ -> False
```

Comment:
These last two lines can be combined into:

```haskell
_ -> False
```

Reply:
Right!